src/test/java/net/sf/jsqlparser/statement/select/SelectTest.java

```diff
@@ -4804,6 +4804,14 @@ public void testCurrentIssue940() throws JSQLParserException {
                 "SELECT date(current) AS test_date FROM systables WHERE tabid = 1");
     }
 
+    @Test
```

Comment:
Please add a test, where you set the modes using plain Java code, one after another.
I would assume, that setting one mode would have to switch off the other modes.